packages/runtime-dom/src/apiCustomElement.ts

```diff
@@ -250,11 +257,17 @@ export class VueElement extends BaseClass {
     const resolve = (def: InnerComponentDef, isAsync = false) => {
       const { props, styles } = def
 
-      // cast Number-type props set before resolve
       let numberProps
       if (props && !isArray(props)) {
         for (const key in props) {
           const opt = props[key]
+
+          // reflect default value
+          if (this._props[key] === undefined && hasOwn(opt, 'default')) {
+            this._setProp(key, opt.default, true, false)
```

Comment:
Doesn't this also need to account for using a function for the `default`?

- [Playground - This PR](https://deploy-preview-10537--vue-sfc-playground.netlify.app/#eNrNk01r20AQhv/KsBfJ4MiHFgpGLTTGhxSahrpHXRRpLG+6X+yHayP03ztaybJjTHLNRWjnfd/Rs9rZlsmSq+zFsSXj0mjr4edxteOihq3VEpJsMa6zfcCkUKOphRq3XOEqOK/lWqBE5aEbQ4O1UNWl6rIhkibyeEeOOxzqyfxWr3T87GwWG2nlPFQIX6HWVegNWWWx9HiyXzedYlpgJnSTVnijlO1LEXrhqtyg/+695c/BE280JT3JlU37Hdr30tEU04Wa4Mkykt8fH+o0KY1JZhk9UdVx3x+Bl82Zd+Td8oYmRCuakbZQAAWrtDRcoP1lPKdeBVtCVHqtFEL/+xFr3gacn+rVDqu/N+ov7tDXCvZk0aHdY8EmzZeW4AZ5vXnEA71PotR1EOR+Q/yNtM/QMw62+6Bqwr7wRdqHONRcNX/c+uBRudOmetDe2UV/wWjCVm9s/Yz7Kfscc4Xq6C9eXCH6h7mrLDceHPpgvtFQxOl/stq4NLaK5ze19UdDi8cgn9GO2JQog/BL+HKGi8d2FdrQoarmdSidnTwAlgisgmSjJYKLZrq3vULgAz2NQb4YgAmVFh6lEXTxaAWQ13wfXwDaduCGLmZJ29mzFOlGKV8MqXxx0Yt1/wHeZ49U)
- [Playground - main](https://play.vuejs.org/#eNrNk01r20AQhv/KsBfJ4MiHFgpGLTTGhxSahrpHXRRpLG+6X+yHayP03ztaybJjTHLNRWjnfd/Rs9rZlsmSq+zFsSXj0mjr4edxteOihq3VEpJsMa6zfcCkUKOphRq3XOEqOK/lWqBE5aEbQ4O1UNWl6rIhkibyeEeOOxzqyfxWr3T87GwWG2nlPFQIX6HWVegNWWWx9HiyXzedYlpgJnSTVnijlO1LEXrhqtyg/+695c/BE280JT3JlU37Hdr30tEU04Wa4Mkykt8fH+o0KY1JZhk9UdVx3x+Bl82Zd+Td8oYmRCuakbZQAAWrtDRcoP1lPKdeBVtCVHqtFEL/+xFr3gacn+rVDqu/N+ov7tDXCvZk0aHdY8EmzZeW4AZ5vXnEA71PotR1EOR+Q/yNtM/QMw62+6Bqwr7wRdqHONRcNX/c+uBRudOmetDe2UV/wWjCVm9s/Yz7Kfscc4Xq6C9eXCH6h7mrLDceHPpgvtFQxOl/stq4NLaK5ze19UdDi8cgn9GO2JQog/BL+HKGi8d2FdrQoarmdSidnTwAlgisgmSjJYKLZrq3vULgAz2NQb4YgAmVFh6lEXTxaAWQ13wfXwDaduCGLmZJ29mzFOlGKV8MqXxx0Yt1/wHeZ49U)

Reply:
Updated. feel free to review.